File: Bridge/Kaliuzhnyi-03/ConsoleApp1/Shapes/Shape.cs
using Bridge.Colors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bridge.Shapes;
public abstract class Shape
{
    protected IColor _color;
    public string Color { get => _color.GetColor(); }
    public Shape(IColor color)
    {
        if (color == null)
        {
            throw new ArgumentNullException("Error! Color can't be null!");
        }
        _color = color;
    }
    public abstract double Perimeter();
    public abstract double Area();
}

File: Bridge/Mozil/shapes/Shape.java
package shapes;

import colors.Color;

public abstract class Shape {
    protected Color color;

    public Shape(Color color) {
        this.color = color;
    }

    protected abstract double countArea();

    protected abstract double countPerimeter();

    @Override
    public String toString() {
        return this.getClass().getSimpleName() +
                " color = " + color +
                " area = " + this.countArea() +
                " perimeter = " + this.countPerimeter();
    }
}
